### PR TITLE
walletconnect

### PR DESCRIPTION
**Pull request by melnicenkovadik**

### CODE REVIEW

**Review by Motzart — APPROVED** [review]

File: src/components/layouts/Header/Header.js [hunk 1]
```diff
@@ -40,6 +40,11 @@ export const Header = observer(() => {
 
   const logOut = async () => {
     deactivate();
+    localStorage.removeItem('connector');
+    if (window.localStorage.getItem('connector') === 'walletconnect') {
+      await walletconnect.close();
+      walletconnect.walletConnectProvider = null;
+    }
```
Comment (mrlika) [inline src/components/layouts/Header/Header.js:47]: Duplicated code. I think this code should be moved to `deactivate` or a reusable function.

File: src/hooks/useInterval.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }, [callback]);
+
+  // Set up the interval.
+  useEffect(() => {
+    function tick() {
+      const { current } = savedCallback;
+      if (current) {
+        current();
+      }
+    }
```
Comment (mrlika) [inline src/hooks/useInterval.js:18]: Can be simplified to `tick() => savedCallback.current?.();`

File: src/hooks/useInterval.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    if (delay !== null) {
```
Comment (mrlika) [inline src/hooks/useInterval.js:20]: No need to run the interval if `savedCallback.current` is `undefined` (considering `if (current)` check above)

**Review by mrlika — CHANGES_REQUESTED** [review]

File: src/components/layouts/Header/Header.js (hunk 1 above)
Comment (melnicenkovadik) [reply src/components/layouts/Header/Header.js:47]: I'm still doing kiss and then I'll put it in the "useAuth" hook

Comment (melnicenkovadik) [reply src/components/layouts/Header/Header.js]: thank

File: src/hooks/useInterval.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      if (leading) tick();
+      const id = setInterval(tick, delay);
+      return () => clearInterval(id);
+    }
+    return undefined;
```
[the remark this answers is not in the document]
Comment (melnicenkovadik) [reply src/hooks/useInterval.js:20]: our eslint config =)

File: src/services/staking.wrapper.js [hunk 5]
```diff
@@ -116,6 +116,7 @@ export default class StakingWrapper {
     const providerOrSigner = loggedIn
       ? new providers.Web3Provider(connector).getSigner()
       : this.privateProvider;
+
```
[the remark this answers is not in the document]
Comment (melnicenkovadik) [reply src/services/staking.wrapper.js:119]: why?

[the remark this answers is not in the document]
Comment (melnicenkovadik) [reply src/services/staking.wrapper.js:119]: are u from Zhytomyr? me too!)
add me to telegram please, i have questions)  - @avokadikvadik

File: src/hooks/useInterval.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from 'react';
+
+export default function useInterval(callback, delay, leading = true) {
```
[the remark this answers is not in the document]
Comment (melnicenkovadik) [reply src/hooks/useInterval.js:3]: thank

**Pull request merged**